packages/stokhos/src/sacado/kokkos/pce/tpetra/Stokhos_Tpetra_ETI_Helpers_UQ_PCE_DEVICE.hpp

```diff
@@ -46,23 +46,43 @@
 #define INSTANTIATE_UQ_PCE_STORAGE(INSTMACRO, STORAGE, LO, GO, N)      \
   INSTMACRO( Sacado::UQ::PCE<STORAGE>, LO, GO, N )
 
+#define INSTANTIATE_UQ_PCE_STORAGE_SD(INSTMACRO, STORAGE, N)      \
+  INSTMACRO( Sacado::UQ::PCE<STORAGE>, N )
+
 #define INSTANTIATE_UQ_PCE_DS_SLD(INSTMACRO, S, L, D, LO, GO, N)       \
   typedef Stokhos::DynamicStorage<L,S,D::execution_space> DS_ ## L ## _ ## S ## _ ## _ ## D; \
   INSTANTIATE_UQ_PCE_STORAGE(INSTMACRO, DS_ ## L ## _ ## S ## _ ## _ ## D, LO, GO, N)
 
+#define INSTANTIATE_UQ_PCE_DS_SLD_SD(INSTMACRO, S, L, D, N)       \
+  typedef Stokhos::DynamicStorage<L,S,D::execution_space> DS_ ## L ## _ ## S ## _ ## _ ## D ## _2; \
+  INSTANTIATE_UQ_PCE_STORAGE_SD(INSTMACRO, DS_ ## L ## _ ## S ## _ ## _ ## D ## _2, N)
+
 #define INSTANTIATE_UQ_PCE_S_D(INSTMACRO, D, LO, GO, N) \
   INSTANTIATE_UQ_PCE_DS_SLD(INSTMACRO, double, int, D, LO, GO, N)
 
+#define INSTANTIATE_UQ_PCE_S_D_SD(INSTMACRO, D, N) \
+  INSTANTIATE_UQ_PCE_DS_SLD_SD(INSTMACRO, double, int, D, N)
+
 #define INSTANTIATE_UQ_PCE_S(INSTMACRO, LO, GO, N) \
   typedef Stokhos::DeviceForNode2<N>::type DFN_ ## LO ## _ ## GO ## _ ## N; \
   INSTANTIATE_UQ_PCE_S_D(INSTMACRO, DFN_ ## LO ## _ ## GO ## _ ## N, LO, GO, N)
 
+#define INSTANTIATE_UQ_PCE_S_SD(INSTMACRO, N) \
+  typedef Stokhos::DeviceForNode2<N>::type DFN_ ## N; \
+  INSTANTIATE_UQ_PCE_S_D_SD(INSTMACRO, DFN_ ## N, N)
+
 #define INSTANTIATE_UQ_PCE(INSTMACRO, LO, GO, N) \
   INSTANTIATE_UQ_PCE_S(INSTMACRO, LO, GO, N)
 
+#define INSTANTIATE_UQ_PCE_SD(INSTMACRO, N) \
+  INSTANTIATE_UQ_PCE_S_SD(INSTMACRO, N)
+
 #define INSTANTIATE_TPETRA_UQ_PCE_N(INSTMACRO, N)  \
   INSTANTIATE_UQ_PCE_S(INSTMACRO, int, int, N)
```

Comment:
@etphipp Does `int, int` here refer to LO, GO?  This might be the reason for https://github.com/trilinos/Trilinos/issues/4902, then.

Reply:
Yes, I think the first one is LO and the second was is GO.  The fix might be as simple as replacing them with `Tpetra::Details::DefaultTypes::local_ordinal_type` and `Tpetra::Details::DefaultTypes::global_ordinal_type`.